tests/rptest/tests/archival_upload_test.py

```diff
@@ -148,7 +148,7 @@ def __init__(self, test_context):
         if self.real_thing:
             extra_rp_conf = dict(
                 developer_mode=True,
-                disable_metrics=False,
+                disable_metrics=True,
```

Comment:
@jcsp i'm not sure there is a reason to disable them here in the context of a ducktape test, and i agree they should be enabled when possible even if not in use.

the one place we do disable them, which you may have already noticed, is in the c++ unit tests because of double metric registration when tests re-use a reactor instance. i think the latest changes to remove globals in seastar is going to eliminate that issue (yay).